### PR TITLE
Test/internal/nosql/cassandra test

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR implements cassandra test.
It also include the following changes:

1. [Refactor] Create `ClusterConfig` mock implementation for the testing
2. [Refactor] Separate `events` struct definition for the testing
3. [Improvement] Avoid runtime instantiation of `WhiteListHostFilter`
4. [Bugfix] Use `dcHost` in `DataCentreHostFilter`

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/db/nosql/cassandra/cassandra_test.go [hunk 1]
```diff
@@ -994,38 +1744,23 @@ func TestDelete(t *testing.T) {
...
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "delete returns qb.delete",
+			args: args{
+				table: "t",
+				cmps: []Cmp{
+					Eq("col"),
```
Comment (hlts2) [inline internal/db/nosql/cassandra/cassandra_test.go:1752]: should we increase the number of `Cmp `  ?

**Review by hlts2 — CHANGES_REQUESTED** [review]

File: internal/db/nosql/cassandra/cassandra_test.go [hunk 2]
```diff
@@ -918,40 +1682,26 @@ func TestSelect(t *testing.T) {
...
-		       }
-		   }(),
-		*/
+		func() test {
+			stmt, names := qb.Select("t").Columns("col").Where(Eq("col")).ToCql()
+			return test{
+				name: "selete return qb.select",
+				args: args{
+					table:   "t",
+					columns: []string{"col"},
```
Comment (hlts2) [inline internal/db/nosql/cassandra/cassandra_test.go:1691]: same 
https://github.com/vdaas/vald/pull/809/files#r517132827

File: internal/db/nosql/cassandra/cassandra_test.go [hunk 3]
```diff
@@ -710,124 +1574,24 @@ func Test_client_Query(t *testing.T) {
...
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "query return gocqlx.Query",
+			args: args{
+				stmt:  "stmt",
+				names: []string{"n"},
```
Comment (hlts2) [inline internal/db/nosql/cassandra/cassandra_test.go:1581]: same
https://github.com/vdaas/vald/pull/809/files#r517132827

File: internal/db/nosql/cassandra/cassandra_test.go [hunk 4]
```diff
@@ -187,122 +1237,37 @@ func Test_client_Open(t *testing.T) {
...
+				ctx: context.Background(),
+			},
+			fields: fields{
+				cluster: &MockClusterConfig{
+					CreateSessionFunc: func() (*gocql.Session, error) {
+						return nil, nil
+					},
+				},
+			},
+			want: want{},
```
Comment (hlts2) [inline internal/db/nosql/cassandra/cassandra_test.go:1252]: could you please check the session object if error or not with checkFunc?

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]

**Pull request merged**